### DISCUSSION TITLE
Any developer documentation?

### DISCUSSION THREAD

**Discussion by OpenShooter** [WinMerge/winmerge#838] [Q&A]
After problems comparing large files without using prediffer, I wondered how difficult it would be to add this capability to WinMerge. I donwloaded the source code and tried to look through it, but I could not find any information about how the code is organized, and the code is barely commented.

Is there any overview documentation, which explains which classes are used for which user function, and where the comparisons are actually done?

**Comment by sdottaka** [comment]
It's out of date, but the developer documentation is below.

https://github.com/WinMerge/winmerge/tree/master/Docs/Developers

You can generate documentation with doxygen from source code, but it may not be very useful

**Reply by OpenShooter** [reply]
The documents you referred to do have nothing that I could find about the organization of the program. The source code consists of 46 projects. What I am looking for (this time) is the function that compares a line from each of 2 files, which I assumed is in the project Merge. But where to look? This is the kind of information I am looking for, and which any developer needs.

**Reply by sdottaka** [reply]
WinMerge uses very old GNU diffutils sources. The file comparison function is diff_2_files () in Src\diffutils\src\analyze.c. If you put a breakpoint here, you will be able to trace the caller.

**Reply by OpenShooter** [reply]
Thanks for the info. I cannot get a break here, because I don't have the `.pdb`, and I cannot generate one because I can't build the solution (I submitted an issue for this on GitHub).
